src/view/screens/NotificationsSettings.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react'
+import {View} from 'react-native'
+import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome'
+import {msg, Trans} from '@lingui/macro'
+import {useLingui} from '@lingui/react'
+import {useMutation, useQueryClient} from '@tanstack/react-query'
+
+import {until} from '#/lib/async/until'
+import {AllNavigatorParams, NativeStackScreenProps} from '#/lib/routes/types'
+import {logger} from '#/logger'
+import {isWeb} from '#/platform/detection'
+import {
+  RQKEY as RQKEY_NOTIFS,
+  useNotificationFeedQuery,
+} from '#/state/queries/notifications/feed'
+import {useAgent} from '#/state/session'
+import * as Toast from '#/view/com/util/Toast'
+import {ViewHeader} from '#/view/com/util/ViewHeader'
+import {CenteredView} from '#/view/com/util/Views'
+import {atoms as a, useTheme} from '#/alf'
+import * as Toggle from '#/components/forms/Toggle'
+import {Loader} from '#/components/Loader'
+import {Text} from '#/components/Typography'
+
+type Props = NativeStackScreenProps<AllNavigatorParams, 'NotificationsSettings'>
+export function NotificationsSettingsScreen({}: Props) {
+  const {data, refetch} = useNotificationFeedQuery()
+  const queryClient = useQueryClient()
+  const agent = useAgent()
+  const priority = data?.pages.at(0)?.priority
+
+  const {_} = useLingui()
+  const t = useTheme()
+
+  const {mutate: onChangePriority, isPending} = useMutation({
+    mutationFn: async (keys: string[]) => {
+      const enabled = keys[0] === 'enabled'
+
+      eagerlySetCachedPriority(queryClient, enabled)
```

Comment:
I assume we should be doing the inverse of this in `onError`. A refetch might not return the correct state since a network error could have caused the failure.